ocean_lib/data_provider/data_service_provider.py

```diff
@@ -498,6 +498,17 @@ def get_root_uri(service_endpoint: str) -> str:
         if not result:
             raise InvalidURL(f"InvalidURL {service_endpoint}.")
 
+        try:
```

Comment:
I think you should pull this logic out into a separate method like `is_provider_valid(provider_uri)`